src/analysis/inner_product_space/positive.lean

```diff
@@ -32,75 +39,157 @@ of requiring self adjointness in the definition.
+    let a : U := ⟨u, hu⟩,
+    let b : V := ⟨v, hv⟩,
+    have hau : u = a := rfl,
+    have hbv : v = b := rfl,
+    rw [hau, ← submodule.linear_proj_of_is_compl_apply_left hUV a,
+      ← submodule.subtype_apply _, ← comp_apply, ← h.1 _ _, comp_apply, hbv,
+      submodule.linear_proj_of_is_compl_apply_right hUV b, map_zero, inner_zero_left], },
+  { intro h,
+    have : (U.subtype.comp (U.linear_proj_of_is_compl V hUV)).is_symmetric,
+    { intros x y,
+      nth_rewrite 0 ← submodule.linear_proj_add_linear_proj_of_is_compl_eq_self hUV y,
+      nth_rewrite 1 ← submodule.linear_proj_add_linear_proj_of_is_compl_eq_self hUV x,
+      simp_rw [inner_add_right, inner_add_left, comp_apply, submodule.subtype_apply _,
+        ← submodule.coe_inner, submodule.is_ortho_iff_inner_eq.mp h _
+          (submodule.coe_mem _) _ (submodule.coe_mem _),
+        submodule.is_ortho_iff_inner_eq.mp h.symm _
+          (submodule.coe_mem _) _ (submodule.coe_mem _)], },
+    refine ⟨this, _⟩,
+    intros x,
+    rw [comp_apply, submodule.subtype_apply, ← submodule.linear_proj_of_is_compl_idempotent,
+      ← submodule.subtype_apply, ← comp_apply, this ((U.linear_proj_of_is_compl V hUV) x) _],
+    exact inner_self_nonneg, },
 end
 
-lemma is_positive_one : is_positive (1 : E →L[𝕜] E) :=
-⟨is_self_adjoint_one _, λ x, inner_self_nonneg⟩
+open_locale nnreal
 
-lemma is_positive.add {T S : E →L[𝕜] E} (hT : T.is_positive)
-  (hS : S.is_positive) : (T + S).is_positive :=
+/-- the spectrum of a positive linear map is non-negative -/
+lemma is_positive.nonneg_spectrum [finite_dimensional 𝕜 E] {T : E →ₗ[𝕜] E} (h : T.is_positive) :
+  spectrum 𝕜 T ⊆ set.range (algebra_map ℝ≥0 𝕜) :=
 begin
-  refine ⟨hT.is_self_adjoint.add hS.is_self_adjoint, λ x, _⟩,
-  rw [re_apply_inner_self, add_apply, inner_add_left, map_add],
-  exact add_nonneg (hT.inner_nonneg_left x) (hS.inner_nonneg_left x)
+  cases h with hT h,
+  intros μ hμ,
+  rw [set.mem_range, nnreal.exists],
+  simp_rw [← module.End.has_eigenvalue_iff_mem_spectrum] at hμ,
+  have : ↑(re μ) = μ,
+  { simp_rw [← eq_conj_iff_re],
+    exact is_symmetric.conj_eigenvalue_eq_self hT hμ, },
+  rw ← this at hμ,
+  simp_rw [hT _] at h,
+  exact ⟨re μ, eigenvalue_nonneg_of_nonneg hμ h, this⟩,
 end
 
+section complex
+
+/-- for spaces `V` over `ℂ`, it suffices to define positivity with
+`0 ≤ ⟪T v, v⟫_ℂ` for all `v ∈ V` -/
+lemma is_positive_iff_complex {V : Type*} [normed_add_comm_group V]
+  [inner_product_space ℂ V] (T : V →ₗ[ℂ] V) :
+  T.is_positive ↔ ∀ v : V, ↑(re ⟪T v, v⟫_ℂ) = ⟪T v, v⟫_ℂ ∧ 0 ≤ re ⟪T v, v⟫_ℂ :=
+by simp_rw [is_positive, is_symmetric_iff_inner_map_self_real, inner_conj_symm,
+     ← eq_conj_iff_re, inner_conj_symm, ← forall_and_distrib]
+
+end complex
+
+section continuous_linear_map
+
+variables [complete_space E] [complete_space F]
+
+open continuous_linear_map
+
+lemma is_positiveL_iff (T : E →L[𝕜] E) :
+  (T : E →ₗ[𝕜] E).is_positive ↔ is_self_adjoint T ∧ ∀ x, 0 ≤ T.re_apply_inner_self x :=
```

Comment:
I don't like the name here. It should to something like `to_linear_map_is_positive_iff`?